container-core/src/test/java/com/yahoo/jdisc/http/server/jetty/HttpServerTest.java

```diff
@@ -456,18 +458,51 @@ public void requireThatNoConnectionHeaderMeansKeepAliveInHttp11KeepAliveDisabled
 
     @Test
     public void requireThatConnectionIsClosedAfterXRequests() throws Exception {
-        final int MAX_KEEPALIVE_REQUESTS = 100;
-        final JettyTestDriver driver = JettyTestDriver.newConfiguredInstance(new EchoRequestHandler(),
-                new ServerConfig.Builder(),
-                new ConnectorConfig.Builder().maxRequestsPerConnection(MAX_KEEPALIVE_REQUESTS));
-        for (int i = 0; i < MAX_KEEPALIVE_REQUESTS - 1; i++) {
+        final int MAX_REQUESTS = 10;
+        Path privateKeyFile = tmpFolder.newFile().toPath();
+        Path certificateFile = tmpFolder.newFile().toPath();
+        generatePrivateKeyAndCertificate(privateKeyFile, certificateFile);
+        ConnectorConfig.Builder connectorConfig = new ConnectorConfig.Builder()
+                .maxRequestsPerConnection(MAX_REQUESTS)
+                .ssl(new ConnectorConfig.Ssl.Builder()
+                        .enabled(true)
+                        .clientAuth(ConnectorConfig.Ssl.ClientAuth.Enum.NEED_AUTH)
+                        .privateKeyFile(privateKeyFile.toString())
+                        .certificateFile(certificateFile.toString())
+                        .caCertificateFile(certificateFile.toString()));
+        ServerConfig.Builder serverConfig = new ServerConfig.Builder()
+                .connectionLog(new ServerConfig.ConnectionLog.Builder().enabled(true));
+        JettyTestDriver driver = JettyTestDriver.newConfiguredInstance(
+                new EchoRequestHandler(),
+                serverConfig,
+                connectorConfig,
+                binder -> {});
+
+        // HTTP/1.1
+        for (int i = 0; i < MAX_REQUESTS - 1; i++) {
             driver.client().get("/status.html")
                     .expectStatusCode(is(OK))
                     .expectNoHeader(CONNECTION);
         }
         driver.client().get("/status.html")
                 .expectStatusCode(is(OK))
                 .expectHeader(CONNECTION, is(CLOSE));
+
+        // HTTP/2
+        try (CloseableHttpAsyncClient client = createHttp2Client(driver)) {
+            String uri = "https://localhost:" + driver.server().getListenPort() + "/status.html";
+            for (int i = 0; i < MAX_REQUESTS - 1; i++) {
+                SimpleHttpResponse response = client.execute(SimpleRequestBuilder.get(uri).build(), null).get();
+                assertEquals(OK, response.getCode());
+            }
+            try {
+                client.execute(SimpleRequestBuilder.get(uri).build(), null).get();
+                fail();
+            } catch (ExecutionException e) {
+                // Note: this is a weakness with Apache Http Client 5; the failed stream/request will not be retried on a new connection
```

Comment:
... !? So what effect will closing the stream have, then? Or do you mean that it results in an exception, rather than an automatic retry?

Reply:
Yes, this is the sad thing with Apache http client 5 and the reason why this is "do not merge".

Reply:
Agreed to merge in offline discussion